quetz/dao.py

```diff
@@ -409,16 +409,29 @@ def create_package_member(self, channel_name, package_name, new_member):
         self.db.commit()
 
     def get_api_keys_with_members(self, user_id):
-        return (
+        user_role_api_keys = (
+            self.db.query(ApiKey)
+            .filter(ApiKey.owner_id == user_id)
+            .outerjoin(Profile, ApiKey.user_id == Profile.user_id)
```

Comment:
probably we don't need to check profile here, because only a real user can be an owner ;-)

Reply:
but we need the inverse of the check above otherwise we'd get API keys twice, right?

Reply:
I may be wrong, but this should be already covered in the condition: `APIKey.user_id == user_id` because `user_id` is an owner, so it must be a real user with a profile. but maybe it's to fragile to depend on. someone might come and make some owners fake users (for example, a team could be an owner), and this will fail. So it might be better to check the profile.

in the future, we could also add a flag to a user model, to check whether it's a real or fake user (such as `User.real = True/False`)

Reply:
I think you're right! Thnaks